src/main/java/org/reactome/release/dataexport/Main.java

```diff
@@ -1,34 +1,44 @@
 package org.reactome.release.dataexport;
 
-import java.io.FileInputStream;
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
 import java.io.IOException;
-import java.util.*;
+
+import org.reactome.release.dataexport.configuration.ConfigurationManager;
 
 /**
  * Generates post-release export files for NCBI, UCSC and Europe PMC.
  * @author jweiser
  */
 public class Main {
+	@Parameter(names={"--overwrite-config-file", "-c"})
+	private boolean overwriteConfigFile;
 
 	/**
 	 * Main method to process configuration file and run the executeStep method of the DataExporterStep class
+	 *
 	 * @param args Command line arguments for the post-release data files export (currently the only argument is,
-	 *     optionally, the configuration file
-	 * @throws IOException Thrown if unable to read configuration file, create output directory or write files
-	 * @throws Exception Thrown if execution of executeStep method in DataExporterStep fails
+	 *     optionally, "--overwrite-config-file" or "-c" to indicate the configuration file should be (re)created
```

Comment:
Why would someone want to overwrite the config file? Does it get overwritten with some default file?

Reply:
The default in the program is not to overwrite, so if the configuration file already exist and is valid, it will be used and the user won't be interactively prompted for anything.

This option allows you to have the configuration file be created from scratch through user obtained input in case the old configuration values are out of date, though.  It is just a convenience of setting up the configuration and running the program with one command, rather than having to modify the configuration file by hand.

Reply:
What does it mean for a configuration to be valid? I don't think I saw a validator in here, but I may have missed it.

Prompting the user for inputs is very nice, but since this will probably be run by Jenkins, I worry that a lot of effort was put into code that is probably not going to be used very much.

Reply:
Yeah, it may not be used as much with Jenkins.  I was just migrating the existing configuration creation code that was present in a separate shell script into the Java code because I wanted to be able to validate the configuration file had everything necessary to both create the files and upload them.  The new uploader classes have the properties they need as part of their implementation and this is used for validation.

In a nutshell, validation checks both the configuration values that are needed for the program as a whole and the configuration values needed for a specific purpose (e.g. accessing the EuropePMC FTP server) are present. 

This is the part of the code that does the validation: https://github.com/reactome/release-data-exporter/pull/4/files#diff-cea3df45b813744c096315851a35f8432345c99f90a57080ca9eb25020efdd38R163-R184.

Reply:
Yes, I guessed I missed that method on the first pass-through.

Reply:
No problem